webapp/src/test/java/io/github/microcks/web/RestControllerIT.java

```diff
@@ -137,4 +152,92 @@ public void testHeadersTemplating() {
       String requestBasedHeader = response.getHeaders().getFirst("x-request-based-header");
       assertEquals("XL size", requestBasedHeader);
    }
+
+   @Test
+   public void testProxyFallback() {
+      // Upload pastry-with-proxy-fallback and pastry-for-proxy specs
+      uploadArtifactFile("target/test-classes/io/github/microcks/util/openapi/pastry-with-proxy-fallback-openapi.yaml",
+            true);
+      uploadArtifactFile("target/test-classes/io/github/microcks/util/openapi/pastry-for-proxy-openapi.yaml", true);
+
+      // Set real port to the dispatcher
+      Service service = serviceRepository.findByNameAndVersion("pastry-proxy", "1.0.0");
+      Operation op = service.getOperations().stream().filter(o -> o.getName().endsWith("GET /pastry/{name}"))
+            .findFirst().orElseThrow();
+      op.setDispatcherRules(op.getDispatcherRules().replaceFirst("http://localhost", getServerUrl()));
+      serviceRepository.save(service);
+
+      ResponseEntity<String> response = restTemplate.getForEntity("/rest/pastry-proxy/1.0.0/pastry/realDonut",
+            String.class);
+      assertEquals(200, response.getStatusCode().value());
+      try {
+         JSONAssert.assertEquals("{\"name\":\"Real One\"}", response.getBody(), JSONCompareMode.LENIENT);
+      } catch (Exception e) {
+         fail("No Exception should be thrown here");
+      }
+   }
+
+   @Test
+   public void testProxyFallbackWithEqualsOriginAndExternalUrls() {
+      // Upload pastry-with-proxy-fallback and pastry-for-proxy specs
+      uploadArtifactFile("target/test-classes/io/github/microcks/util/openapi/pastry-with-proxy-fallback-openapi.yaml",
+            true);
+      uploadArtifactFile("target/test-classes/io/github/microcks/util/openapi/pastry-for-proxy-openapi.yaml", true);
+
+      // Set original URL to the dispatcher
+      Service service = serviceRepository.findByNameAndVersion("pastry-proxy", "1.0.0");
+      Operation op = service.getOperations().stream().filter(o -> o.getName().endsWith("GET /pastry/{name}"))
+            .findFirst().orElseThrow();
+      op.setDispatcherRules(op.getDispatcherRules().replaceFirst("http://localhost", getServerUrl())
+            .replaceFirst("pastry-real", "pastry-proxy"));
+      serviceRepository.save(service);
+
+      ResponseEntity<String> response = restTemplate.getForEntity("/rest/pastry-proxy/1.0.0/pastry/realDonut",
```

Comment:
I think I understand how `/pastry/realDonut` can be resolved by the proxy...
As `pastry-for-proxy-openapi.yaml` only defines a `/pastry/donut` request path, it actually not match and rely on the fact that Microcks will still answer with the only existing question...

It would have been better IMHO to have something like:
`/pastry/donut` -> not found then redirect to proxy -> `/pastry/donut` -> found on proxy

I will probably adapt a bit this sample to that test will not rely on Microcks mechanism for the backend and will be easier understandable

Reply:
Yes, I agree that this example is not clear enough. Its main task is to verify that if someone defines the base URL of mitchrocks, it will not result in a circular query. To simulate this, I change the path in [line 192](https://github.com/microcks/microcks/pull/1101/files/c20afb0eb16e37d548b75561f073cefaa7ec2475#diff-d7164af73de0fcfd4aba7238940c3655bb45e4b702f76316e3118f7428448cd9R192). But, of course, it's not the best solution.